File: aemet/Aemet-Java-IvanRC/src/main/java/dto/MapaMedicionesDto.java
package dto;

import com.squareup.moshi.Json;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.simpleframework.xml.ElementMap;
import org.simpleframework.xml.Root;

import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Json(name = "mapa_mediciones")
@Root(name = "mapa_mediciones")
public class MapaMedicionesDto {

    @ElementMap(name ="mapa_mediciones", inline = true)
    Map<String, ListaMedicionesDto> mapaMediciones;
}

File: aemet/Aemet-kotlin-IvanRC/src/main/kotlin/dto/MapaMedicionesDto.kt
package dto

import com.squareup.moshi.Json
import org.simpleframework.xml.ElementMap
import org.simpleframework.xml.Root

@Json(name = "mapa_mediciones")
@Root(name = "mapa_mediciones")
data class MapaMedicionesDto (
    @Json(name = "mapa_mediciones")
    @field:ElementMap(name = "mapa_mediciones", inline = true)
    @param:ElementMap(name = "mapa_mediciones", inline = true)
    val medicionesDto: Map<String, ListaMedicionesDto>
)